tests/python_package_test/test_engine.py

```diff
@@ -4398,3 +4393,51 @@ def test_quantized_training():
     quant_bst = lgb.train(bst_params, ds, num_boost_round=10)
     quant_rmse = np.sqrt(np.mean((quant_bst.predict(X) - y) ** 2))
     assert quant_rmse < rmse + 6.0
+
+
+@pytest.mark.parametrize("use_weight", [False, True])
+@pytest.mark.parametrize("num_boost_round", [5, 15])
+@pytest.mark.parametrize(
+    "custom_objective, objective_name, df, num_class",
+    [
+        (mse_obj, "regression", make_synthetic_regression(), 1),
+        (
+            multiclass_custom_objective,
+            "multiclass",
+            make_blobs(n_samples=100, centers=[[-4, -4], [4, 4], [-4, 4]], random_state=42),
+            3,
+        ),
+    ],
+)
+@pytest.mark.skipif(getenv("TASK", "") == "cuda", reason="Skip due to ObjectiveFunction not exposed for cuda devices.")
+def test_objective_function_class(use_weight, num_boost_round, custom_objective, objective_name, df, num_class):
+    X, y = df
+    rng = np.random.default_rng()
+    weight = rng.choice([1, 2], y.shape) if use_weight else None
+    lgb_train = lgb.Dataset(X, y, weight=weight, init_score=np.zeros((len(y), num_class)))
+
+    params = {
+        "verbose": -1,
+        "objective": objective_name,
+        "num_class": num_class,
+        "device": "cpu",
+    }
+    builtin_loss = builtin_objective(objective_name, copy.deepcopy(params))
+    builtin_convert_outputs = lgb.ObjectiveFunction(objective_name, copy.deepcopy(params)).convert_outputs
+
+    params["objective"] = builtin_loss
+    booster_exposed = lgb.train(params, lgb_train, num_boost_round=num_boost_round)
+
+    params["objective"] = objective_name
+    booster = lgb.train(params, lgb_train, num_boost_round=num_boost_round)
+
+    params["objective"] = custom_objective
+    booster_custom = lgb.train(params, lgb_train, num_boost_round=num_boost_round)
+
+    np.testing.assert_allclose(booster_exposed.predict(X), booster.predict(X, raw_score=True))
```

Comment:
Just to understand this 100%: if you're using a builtin objective function, it automatically applies `ConvertOutputs` unless you specify `raw_score=True`? That means that, if you're using a custom objective (i.e. pass a function to the "objective" parameter), `raw_score=True` does not have any effect?

Reply:
Yes that is the case.